File: cars-challenge/cars.js
const paintColors = ['red', 'orange', 'yellow', 'green', 'blue', 'purple', 'black'];

const calculateColor = (daysSinceStart, daysOut) => {
    const sinceProdStart = daysSinceStart + daysOut;
    if (sinceProdStart % 7 === 5 || sinceProdStart % 7 === 6) {
        return "no cars today";
    };
    let workingDays = 0;
    for (let i = 0; i <= sinceProdStart; i++) {
        if(i % 7 === 5 || i % 7 === 6) {
            workingDays = workingDays
        }else{
            workingDays++;
        };
    };
    let colorIndex = (workingDays % 7) - 1;
    if (colorIndex < 0) {
        colorIndex += 7
    }
    return paintColors[colorIndex];
}
console.log(calculateColor(0, 0));
console.log(calculateColor(0, 1));
console.log(calculateColor(0, 2));
console.log(calculateColor(0, 3));
console.log(calculateColor(0, 4));
console.log(calculateColor(0, 5));
console.log(calculateColor(0, 6));
console.log(calculateColor(0, 7));
console.log(calculateColor(0, 8));
console.log(calculateColor(0, 9));
console.log(calculateColor(0, 10));
console.log(calculateColor(0, 11));
console.log(calculateColor(0, 12));
console.log(calculateColor(0, 13));
console.log(calculateColor(0, 14));
console.log(calculateColor(0, 15));
console.log(calculateColor(0, 16));

File: cars-challenge/cars.rb
@colors = ['red','orange','yellow','green','blue','purple','black']
def find_color days_since, days_out
  days = days_since + days_out
  week_day = days % 7
  if week_day == 5
    puts "No cars, that is a Saturday"
  elsif week_day == 6
    puts "No cars, that is a Sunday"
  else
    skipped = ((days - week_day) / 7) * 2
    working = days - skipped
    puts @colors[working % 7]
  end
end
find_color 0, 0
find_color 0, 1
find_color 0, 2
find_color 0, 3
find_color 0, 4
find_color 0, 5
find_color 0, 6
find_color 0, 7
find_color 0, 8
find_color 0, 9
find_color 0, 10
find_color 0, 11
find_color 0, 12